examples/match_statements/src/main.sw

```diff
@@ -0,0 +1,26 @@
+script;
+
+fn main() -> u64 {
+    let x = 5;
+
+    // Match as an expression.
+    let a = match 8 {
+        7 => { 4 },
+        9 => { 5 },
+        8 => { 6 },
+        _ => { 100 },
+    };
+
+    // Match as an expression.
+    let b = match x {
+        5 => { 42 },
+        _ => { 24 },
+    };
+
+    // Match as control flow.
+    match 42 {
```

Comment:
This is also an expression since it returns its value. A more "pure" control flow example could be:

```
fn main() {
  let boolean = true;
  match boolean {
    true => {
        foo();
    },
    false =>  {
      bar();
    }
  };
}

fn foo() {
  // do something
}

fn bar() {
  // do_something
}
```

because the returned value from `match` is truly never used and is therefore only used for control flow

Reply:
Ah, yeah you are right. I'll revise.